### PR TITLE
 fix(): empty namespace of involvedObject fixed

### PR DESCRIPTION
**Pull request by gourishkb**
when involvedObject has an empty namespace (ex: namespace object)
we use the name of the object in the place of namespace.
This will fix the issue of creating an event key with
empty value for namespace field used in caching and aggregation.

### CODE REVIEW

File: pkg/events/events_test.go [hunk 1]
```diff
@@ -165,3 +166,32 @@ func TestRecordEventWithCache(t *testing.T) {
...
+	err := recorder.RecordEvent(context.Background(), &events.Event{
+		Object:            pod,
+		RelatedObject:     nil,
+		ReportingInstance: "controller",
+		Name:              events.EventExampleEvent,
+	})
+	assert.Nil(t, err)
+
+	e := clientMock.createdObject.(*corev1.Event)
+	assert.NotEqual(t, e.InvolvedObject.Name, "")
```
Comment (gourishkb) [inline pkg/events/events_test.go:194]: @rahulsawra98 : "you should avoid using conditionals in test as much as possible, instead use asserts"

addressed here

**Review by rahulsawra98 — APPROVED** [review]

File: pkg/events/events_test.go [hunk 2]
```diff
@@ -165,3 +166,32 @@ func TestRecordEventWithCache(t *testing.T) {
...
+			Name: "test-pod",
+		},
+	}
+	err := recorder.RecordEvent(context.Background(), &events.Event{
+		Object:            pod,
+		RelatedObject:     nil,
+		ReportingInstance: "controller",
+		Name:              events.EventExampleEvent,
+	})
+	assert.Nil(t, err)
```
Comment (kmjayadeep) [inline pkg/events/events_test.go:191]:
```suggestion
	assert.NoError(t, err)
```

File: pkg/events/events_test.go [hunk 3]
```diff
@@ -165,3 +166,32 @@ func TestRecordEventWithCache(t *testing.T) {
...
+		Cluster:   "cluster-1",
+		Component: "controller",
+		Namespace: "test",
+	})
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-pod",
+		},
+	}
```
Comment (kmjayadeep) [inline pkg/events/events_test.go:184]: This is not a good example for test. In Pod spec, namespace is a mandatory field.
Were you able to fix the original issue with Namespace object with this fix? I'm not sure how we can raise events for cluster scoped objects like namespaces or clusterrole.

**Review by kmjayadeep — CHANGES_REQUESTED** [review]

File: pkg/events/events_test.go (hunk 3 above)
Comment (gourishkb) [reply pkg/events/events_test.go]: This change fixed the issue while testing locally. Using namespace object instead now.

**Pull request merged**